foreman/data_refinery_foreman/foreman/management/commands/rerun_salmon_old_samples.py

```diff
@@ -0,0 +1,108 @@
+
+logger = get_and_configure_logger(__name__)
+
+def update_salmon_versions(experiment: Experiment):
+    quant_results = get_quant_results_for_experiment(experiment)
+    salmon_versions = list(quant_results.order_by('-organism_index__created_at')\
+                                   .values_list('organism_index__salmon_version', flat=True)\
+                                   .distinct())
+    
+    if len(salmon_versions) <= 1:
+        # only apply this command on experiments that have more than one salmon version applied on their samples
+        return
+
+    latest_salmon_version = salmon_versions[0]
+
+    # find the samples that were not processed with `latest_salmon_version` and trigger new processor jobs for them
+    newest_computational_results = ComputationalResult.objects.all()\
+        .filter(
+            samples=OuterRef('id'),
+            processor__name=ProcessorEnum.SALMON_QUANT.value['name']
+        )\
+        .order_by('-created_at')
+    
+    samples = experiment.samples.all().annotate(
+            salmon_version=Subquery(newest_computational_results.values('organism_index__salmon_version')[:1])
+        )\
+        .exclude(salmon_version=latest_salmon_version)
+
+    # create new processor jobs for the samples that were run with an older salmon version
+    for sample in samples:
+        original_files = list(sample.original_files.all())
+
+        if not len(original_files): continue
+
+        # Ensure that there's no processor jobs for these original files that the foreman
+        # might want to retry (failed | hung | lost)
+        has_open_processor_job = ProcessorJob.objects.all()\
+                                    .filter(original_files = original_files[0], pipeline_applied=ProcessorPipeline.SALMON)\
+                                    .filter(
+                                        Q(success=False, retried=False, no_retry=False) | 
+                                        Q(success=None, retried=False, no_retry=False, start_time__isnull=False, end_time=None, nomad_job_id__isnull=False) |
+                                        Q(success=None, retried=False, no_retry=False, start_time=None, end_time=None)
+                                    )\
+                                    .exists()
+        if (has_open_processor_job):
+            continue
+
+        create_processor_job_for_original_files(original_files)
+
+def update_salmon_all_experiments():
+    """Creates a tximport job for all eligible experiments."""
+    eligible_experiments = Experiment.objects.all()\
+        .filter(technology='RNA-SEQ', num_processed_samples=0)\
+    
+    for experiment in eligible_experiments:
+        update_salmon_versions(experiment)
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--accession-code",
+            type=str,
+            help=("Optional parameter to allow running the command on a single experiment")
```

Comment:
Oh good call, this'll be nice for making sure it works before we do it to everything!